run.cc

```diff
@@ -60,20 +65,240 @@ void runTests() {
+
+    vector<Data> Dataset;
+
+    string Datenliste[4] = {"exp_A", "exp_B", "exp_C", "exp_D"};
+
+    for (int i=0; i<4; i++)
+    {
+    Data file(Datenliste[i]);
+    Dataset.push_back(file);
+    }
+
+
+    cout << "******************************************************" << endl;
+    runTests();
+    cout << "******************************************************" << endl;
+    // create an object which holds data of experiment A
+
+    Data datA("exp_A");
+    Data datB("exp_B");
+    Data datC("exp_C");
+    Data datD("exp_D");
+
+    //Data& rdatA = datA;
+    //Data& rdatB = datB;
+
+
+    // here is the data from experiment A
+    cout << "bin 27: from " << datA.binLow(27) << " to " << datA.binHigh(27)
+         << endl;
+    cout << "measurement of experiment A in bin 27: " << datA.measurement(27)
+         << endl;
+    cout << "measurement of experiment A in bin 27: " << datA.error(27)
+        << endl;
+    cout << "******************************************************" << endl;
+    cout << "Aufgabe 1c)" << endl;
+
+    cout << "measurement of experiment A in bin 27:" << Dataset[0].measurement(27) << endl;
+    cout << "measurement of experiment B in bin 27:" << Dataset[1].measurement(27) << endl;
+    cout << "measurement of experiment C in bin 27:" << Dataset[2].measurement(27) << endl;
+    cout << "measurement of experiment D in bin 27:" << Dataset[3].measurement(27) << endl;
+
+
+    double a = Dataset[0].measurement(27) - Dataset[1].measurement(27);
+    double b = 2*sqrt(pow(Dataset[0].error(27), 2) + pow(Dataset[1].error(27), 2));
+    double c = b- a;
+
+    cout << "bin 27, c: " << c << endl;
+
+    cout << "******************************************************" << endl;
+    cout << "Aufgabe 1d)" << endl;
+
+    //checkCompatibility(datA, datD, 2);
+    cout << "DatA/DatB" << endl;
+    datA.checkCompatibility(datB, 2);
+    cout << "DatA/DatC" << endl;
+    datA.checkCompatibility(datC, 2);
+    cout << "DatA/DatD" << endl;
+    datA.checkCompatibility(datD, 2);
+    cout << "DatB/DatC" << endl;
+    datB.checkCompatibility(datC, 2);
+    cout << "DatB/DatD" << endl;
+    datB.checkCompatibility(datD, 2);
+
+
+    cout << "******************************************************" << endl;
+    cout << "Aufgabe 1e)" << endl;
+    //cout << "DatA/DatD" << endl;
+
+    Data datav = average(datA,datB);
+    cout << "measurement of average of a and b in bin 27: " << datav.measurement(27)
        << endl;
 
-  return 0;
+    //average(datA, datD);
+
+    cout << "******************************************************" << endl;
+    cout << "Aufgabe 2b)" << endl;
+
+    datA.x2test();
+    datB.x2test();
+    datC.x2test();
+    datD.x2test();
+
+    cout << "******************************************************" << endl;
+    cout << "Aufgabe 2a)" << endl;
+
+    datA.check(2);
+    datB.check(2);
+    datC.check(2);
+    datD.check(2);
+
+    datA.check(3);
+    datB.check(3);
+    datC.check(3);
+    datD.check(3);
+
```

Comment:
In this task you were supposed to compare two datasets with each other, not conduct a bin-wise comparison of a dataset with the fit function ("How many datapoints differ by more than 2 or 3 standard deviations when comparing the measurements of the four datasets?"). So the task in 1d) was to _implement_ the checkCompatibility method, and then in 2a) you're supposed to actually use this function to compare the datasets with each other. Interestingly, compared to your checkCompatibility method, your check method gets the absolute value and the "n" right :) I'll still give you half a point for that.

Also, you didn't answer the question how many deviations you would expect if the data were compatible.